client/etcd/client.go

```diff
@@ -244,7 +244,7 @@ func (c *csclient) etcdClientGen(zone string) (*clientv3.Client, error) {
 }
 
 func newClient(endpoints []string, certfile string, keyfile string, ca string) (*clientv3.Client, error) {
-	tlscfg := &tls.Config{}
+	var tlscfg *tls.Config
 	if certfile != "" {
```

Comment:
Would suggest moving this block to an `initTLSConfig` method since that's pretty isolated from the rest of the stuff.

Reply:
yeah that's the refactoring we were talking about, I'll be refactoring this part later.